templates/template_processor.go

```diff
@@ -0,0 +1,135 @@
+package templates
+
+import (
+	"text/template"
+	"bytes"
+	"github.com/gruntwork-io/boilerplate/errors"
+	"os"
+	"path/filepath"
+	"github.com/gruntwork-io/boilerplate/config"
+	"github.com/gruntwork-io/boilerplate/util"
+	"io/ioutil"
+	"path"
+)
+
+// Copy all the files and folders in templateFolder to outputFolder, passing text files through the Go template engine
+// with the given set of variables as the data.
+func ProcessTemplateFolder(templateFolder string, outputFolder string, variables map[string]string) error {
+	util.Logger.Printf("Processing templates in %s and outputting generated files to %s", templateFolder, outputFolder)
+
+	if err := os.MkdirAll(outputFolder, 0777); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return filepath.Walk(templateFolder, func(path string, info os.FileInfo, err error) error {
+		if shouldSkipPath(path, templateFolder) {
+			util.Logger.Printf("Skipping %s", path)
+			return nil
+		} else if util.IsDir(path) {
+			return createOutputDir(path, templateFolder, outputFolder)
+		} else {
+			return processFile(path, templateFolder, outputFolder, variables)
+		}
+	})
+}
+
+// Copy the given path, which is in the folder templateFolder, to the outputFolder, passing it through the Go template
+// engine with the given set of variables as the data if it's a text file.
+func processFile(path string, templateFolder string, outputFolder string, variables map[string]string) error {
+	isText, err := util.IsTextFile(path)
```

Comment:
Any reason why you opted to put `IsTextFile()` in `util` instead of this package?